### PR TITLE
[flake8] Resolving Q??? errors

### PR DESCRIPTION
**Pull request by john-bodley**
This is the final PR related to remedying the flake8 errors.  This resolves the flake8 Q??? violation errors which enforces single quotes for strings. 

Previously there was a large mix of single and double quotes used for strings, and per PEP8, 

> In Python, single-quoted strings and double-quoted strings are the same. This PEP does not make a recommendation for this. Pick a rule and stick to it. When a string contains single or double quote characters, however, use the other one to avoid backslashes in the string. It improves readability.
> 
> For triple-quoted strings, always use double quote characters to be consistent with the docstring convention in PEP 257.
> 
it mentions to pick a rule and stick to it. Given that the majoring of strings were single-quoted I used that as the default. Note some strings include `'` which I've escaped as opposed to using the other to avoid backslashes. I'm happy to remedy this if necessary. 

to: @mistercrunch

### CODE REVIEW

**Comment by mistercrunch** [comment]
Agreed, single over double quotes.

Personally I prefer switching to avoid backslashes but I feel bad making you make lots of edits.

One thing though is that I prefer not enforcing rules that linters cannot enforce, so can we get pylint to enforce it? If not we'll have to tell each person on their first PR which is annoying both on the reviewer and reviewee side...

**Comment by mistercrunch** [comment]
Looks like this would do it:
https://github.com/edaniszewski/pylint-quotes

**Comment by john-bodley** [comment]
@mistercrunch I can switch from single to double quotes for the strings which include quotes to avoid backslashes. I'll comment on the PR when the change is made.

**Comment by mistercrunch** [comment]
Sweet that will be ideal. It's also consistent with how the JS linter is set up.

**Comment by mistercrunch** [comment]
Now you're in a serious position to win the "Most update on Github" trophy for this week
trophe

**Comment by john-bodley** [comment]
@mistercrunch `flake8` is enforcing the quoting as well; https://github.com/apache/incubator-superset/blob/master/tox.ini#L59.

**Comment by john-bodley** [comment]
@mistercrunch I've update the PR  to remove all occurrences of backslashes in the string.

**Comment by john-bodley** [comment]
Woot!

File: superset/db_engine_specs.py [hunk 1]
```diff
@@ -1043,22 +1043,22 @@ class BQEngineSpec(BaseEngineSpec):
...
+        Grain('quarter', _('quarter'), 'TIMESTAMP_TRUNC({col}, QUARTER)'),
+        Grain('year', _('year'), 'TIMESTAMP_TRUNC({col}, YEAR)'),
     )
 
     @classmethod
     def convert_dttm(cls, target_type, dttm):
         tt = target_type.upper()
         if tt == 'DATE':
-            return "'{}'".format(dttm.strftime('%Y-%m-%d'))
+            return "{}'".format(dttm.strftime('%Y-%m-%d'))
```
Comment (mistercrunch) [inline superset/db_engine_specs.py]: This caused: https://github.com/apache/incubator-superset/issues/4915

File: superset/db_engine_specs.py [hunk 2]
```diff
@@ -1068,21 +1068,21 @@ class ImpalaEngineSpec(BaseEngineSpec):
...
+        Grain('quarter', _('quarter'), "TRUNC({col}, 'Q')"),
+        Grain('year', _('year'), "TRUNC({col}, 'YYYY')"),
     )
 
     @classmethod
     def convert_dttm(cls, target_type, dttm):
         tt = target_type.upper()
         if tt == 'DATE':
-            return "'{}'".format(dttm.strftime('%Y-%m-%d'))
+            return "{}'".format(dttm.strftime('%Y-%m-%d'))
```
Comment (mistercrunch) [inline superset/db_engine_specs.py]: This caused: https://github.com/apache/incubator-superset/issues/4915

**Pull request merged**